monorepo-migration/convert-to-hg.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/sh
+
+set -e
+set -x
+
+CINNABAR=https://github.com/jcristau/git-cinnabar
+CINNABAR_BRANCH=detect-more-renames
+SOURCE_REPO=https://github.com/mozilla-mobile/firefox-android
+SOURCE_REPO_NAME=${SOURCE_REPO##*/}
+SOURCE_BRANCH=main
+DEST_REPO="$(readlink -f "$1")"
+DEST_SUBDIR=mobile/android
+EXPRESSIONS_FILE_PATH="$(dirname $(readlink -f "$0"))/data/message-expressions.txt"
+
+TEMPDIR=$(mktemp -d)
+#trap 'rm -rf ${TEMPDIR} EXIT'
+CINNABAR_PATH="${TEMPDIR}/git-cinnabar"
+SOURCE_REPO_PATH="${TEMPDIR}/${SOURCE_REPO_NAME}"
+
+setup_temporary_repo() {
+  git clone "${SOURCE_REPO}" "${SOURCE_REPO_PATH}"
+  git -C "${SOURCE_REPO_PATH}" config user.email=release@mozilla.com --config user.name="Mozilla Release Engineering"
+}
+
+setup_cinnabar() {
+  git clone --branch "${CINNABAR_BRANCH}" --depth 1 "${CINNABAR}" "${CINNABAR_PATH}"
+  make -C "${CINNABAR_PATH}"
+}
+
+rewrite_git_history() {
+  # replace octopus merges with a series of two-parent merges
+  FILTER_BRANCH_SQUELCH_WARNING=1 \
+  git -C "${SOURCE_REPO_PATH}" filter-branch \
+    --force \
+    --commit-filter \
+      'if test "$#" -lt 7; then
+         git commit-tree "$@"
+       else
+         commit_message=$(mktemp)
+         cat > $commit_message
+             
+         final_tree="$1"
+         shift
+         shift
+         p1="$1"
```

Comment:
Question: does `p1` stands for `parent_revision_1`?

Reply:
It stands for first parent, yes.

Reply:
Okay, let's rename `p1` into `parent1` to be more explicit, then.

Reply:
mercurial very commonly calls them p1/p2, including in user-facing revsets and templates, so I'd prefer to keep it that way.

Reply:
All right, sounds good to me.